### PR TITLE
[5.0] b/c plugin - Load class aliases earlier and move es5 assets

### PR DESCRIPTION
**Pull request by HLeithner**
This PR moves the class alias loading to the constructor function of the plugin to load the alias at the earliest possible execution point.

Additional this PR moves all es5 assets from all joomla.asset.json into the es5.asset.json and make loading them optional.

### Summary of Changes
* Move class aliasing to the constructor
* Move es5 webassets to the plugin

### Testing Instructions
Full CMS with and without plugin active

### Actual result BEFORE applying this Pull Request
Should work

### Expected result AFTER applying this Pull Request
Should still work

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
Do you have anything to say @SharkyKZ or would you only like to spread bad vibes?

File: administrator/language/en-GB/plg_system_compat.ini [hunk 1]
```diff
@@ -5,5 +5,7 @@
 
 PLG_COMPAT_FIELD_CLASSES_ALIASES_LABEL="Classes Aliases"
 PLG_COMPAT_FIELD_CLASSES_ALIASES_DESCRIPTION="Add class aliases for classes which have been renamed or moved to a namespace."
+PLG_COMPAT_FIELD_ES5_ASSETS_LABEL="ES5 Assets"
+PLG_COMPAT_FIELD_ES5_ASSETS_DESCRIPTION="Load the es5 webasset stubs."
```
Comment (brianteeman) [inline administrator/language/en-GB/plg_system_compat.ini]: Should this string be less technical and geeky?

Comment (brianteeman) [reply administrator/language/en-GB/plg_system_compat.ini]: Perhaps explaining when/why you might need to enable it

File: plugins/system/compat/compat.xml [hunk 2]
```diff
@@ -33,6 +33,18 @@
 					<option value="1">JYES</option>
 					<option value="0">JNO</option>
 				</field>
+				<field
+					name="es5_assets"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="PLG_COMPAT_FIELD_ES5_ASSETS_LABEL"
+					description="PLG_COMPAT_FIELD_ES5_ASSETS_DESCRIPTION"
```
Comment (brianteeman) [inline plugins/system/compat/compat.xml]:
```suggestion
					name="es5_assets"
					type="radio"
					label="PLG_COMPAT_FIELD_ES5_ASSETS_LABEL"
					description="PLG_COMPAT_FIELD_ES5_ASSETS_DESCRIPTION"
					layout="joomla.form.field.radio.switcher"
```

File: administrator/language/en-GB/plg_system_compat.ini (hunk 1 above)
Comment (HLeithner) [reply administrator/language/en-GB/plg_system_compat.ini]: I thought about it and I didn't came to a better text, in the end you have to understand what es5 javascript files are... so it's more a template developer setting and not really for end user useful. But I'm happy about a better description

File: plugins/system/compat/src/Extension/Compat.php [hunk 3]
```diff
@@ -41,27 +43,60 @@ public static function getSubscribedEvents(): array
...
+     * @param Event $event
+     * @return void
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    public function onAfterInitialiseDocument(AfterInitialiseDocumentEvent $event)
+    {
+        /**
+         * Load the es5 assets stubs, they are needed if an extension
+         * directly use a core es5 asset which has no function in Joomla 5+
```
Comment (brianteeman) [inline plugins/system/compat/src/Extension/Compat.php]:
```suggestion
         * directly uses a core es5 asset which has no function in Joomla 5+
```

File: administrator/language/en-GB/plg_system_compat.ini (hunk 1 above)
Comment (brianteeman) [reply administrator/language/en-GB/plg_system_compat.ini]: Maybe something along the lines of "This should be enabled if you get an error message saying ..."

**Comment by brianteeman** [comment]
I see that there are still es5 references in 
administrator\templates\atum\joomla.asset.json
build\media_source\com_joomlaupdate\joomla.asset.json
build\media_source\plg_system_jooa11y\joomla.asset.json

**Comment by HLeithner** [comment]
> I see that there are still es5 references in administrator\templates\atum\joomla.asset.json build\media_source\com_joomlaupdate\joomla.asset.json build\media_source\plg_system_jooa11y\joomla.asset.json

Thanks, don't know why my initial search didn't found the 3 assets.... in the build folder... the template I didn't checked tbh

**Comment by brianteeman** [comment]
I just grepped for es5 in all joomla.asset.json files

**Comment by brianteeman** [comment]
Does this need changing as well

https://github.com/joomla/joomla-cms/blob/e7e1fb4f5dbe06b1688a38f9adcea1550f786605/build/build-modules-js/settings.json#L77-L87

**Comment by HLeithner** [comment]
> Does this need changing as well
> 
> https://github.com/joomla/joomla-cms/blob/e7e1fb4f5dbe06b1688a38f9adcea1550f786605/build/build-modules-js/settings.json#L77-L87

I'm not sure I have seen it, @Fedik @dgrammatiko ?

**Comment by HLeithner** [comment]
thanks moved it to the plugin

File: administrator/language/en-GB/plg_system_compat.ini [hunk 4]
```diff
@@ -5,5 +5,7 @@
 
 PLG_COMPAT_FIELD_CLASSES_ALIASES_LABEL="Classes Aliases"
 PLG_COMPAT_FIELD_CLASSES_ALIASES_DESCRIPTION="Add class aliases for classes which have been renamed or moved to a namespace."
+PLG_COMPAT_FIELD_ES5_ASSETS_LABEL="ES5 Assets"
+PLG_COMPAT_FIELD_ES5_ASSETS_DESCRIPTION="Activate this option in case your extension requires *.es5 assets which lead to an exception. The assets provided are empty but prevent the exception."
```
Comment (brianteeman) [inline administrator/language/en-GB/plg_system_compat.ini]: Now I understand it - how is this?

```suggestion
PLG_COMPAT_FIELD_ES5_ASSETS_DESCRIPTION="Activate this option if your extension requires *.es5 assets which has resulted in an exception. The assets provided are empty but prevent the exception."
```

**Comment by HLeithner** [comment]
Hopefully this solve all the JPlugin issues, merging it to get it into alpha3

**Pull request merged**